src/fides/api/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -590,6 +618,8 @@ def infer_target_identity_type(
         target_identity_type = ProvidedIdentityType.email.value
     elif identity_data.phone_number:
         target_identity_type = ProvidedIdentityType.phone_number.value
+    elif identity_data.external_id:
+        target_identity_type = ProvidedIdentityType.external_id.value
```

Comment:
I was a little confused in testing but I think this behavior is okay? I feel like it is what it is, trying to maintain multiple workflows for saving consent in the privacy center.

Consent requests from the privacy center have traditionally been saved against one identity type. It was originally just an email or phone, but then we had a customer request to allow this privacy center consent request to work with only the fides user device id. We prioritized email or phone over fides user device id if they had been provided/verified, but the fides user device id was a catch-all
It seems possible both the external id and fides user device id are supplied. I think it makes sense to prioritize the external id if it exists over the fides user device id as you've done here. But it's important to keep in mind that only one ProvidedIdentity is created. Only the external id would be saved against the consent request.
This one provided identity takes priority if subsequent requests are sent to save privacy requests or notices served with other identities.
So potentially confusing scenario:

Create a consent request with both an external id and a fides user device. Causes a ProvidedIdentity to be created for just external id
Save notices served under the same consent id:
If just fides user device id is in the request body, external id will be magically added in from the "consent request"
If identities are empty, external id alone will show up against the preferences, even though the fides user device was in the original consent request
If external id is different now in request body, it will be overridden by the external id in the original consent request
What do you think?

Reply:
good code comment -